worlds/hk/__init__.py

```diff
@@ -149,23 +158,42 @@ def generate_early(self):
         for option_name in disabled:
             getattr(world, option_name)[self.player].value = 0
 
+    def white_palace_exclusions(self):
+        exclusions = set()
+        wp = self.world.WhitePalace[self.player]
+        if wp <= WhitePalace.option_nopathofpain:
+            exclusions.update(path_of_pain_locations)
+        if wp <= WhitePalace.option_kingfragment:
+            exclusions.update(white_palace_checks)
+        if wp == WhitePalace.option_exclude and self.world.RandomizeCharms[self.player]:
+            # Ensure KF location is still reachable if charms are non-randomized
```

Comment:
The comment on line 169 says to do something if charms are _non-randomized_ but the logic of the if does stuff if they _are randomized_. Is this intentional?

Reply:
Took me a moment to untangle my own logic here, but this is correct:

If White Palace is excluded, we only actually go ahead and exclude the transitions/events/etc related to King Fragment if charms are randomized.  If they are not randomized, then King Fragment is required for most goals and needs to be logically reachable in e.g. ALR or beatable only accessibility -- so we do not exclude transitions or events in WP or the King Fragment location.